Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Driver/Version1/Query/FunctionScoreQuery.php

```diff
@@ -0,0 +1,114 @@
+<?php
+namespace Flowpack\ElasticSearch\ContentRepositoryAdaptor\Driver\Version1\Query;
+
+/*
+ * This file is part of the Flowpack.ElasticSearch.ContentRepositoryAdaptor package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Flowpack\ElasticSearch\ContentRepositoryAdaptor\Exception;
+use TYPO3\Flow\Utility\Arrays;
+
+class FunctionScoreQuery extends FilteredQuery
```

Comment:
Would be great, to have a sentence here, for which use case this query type is used.
Also mention the types somewhere in the readme?